tests/publish/samples.js

```diff
@@ -0,0 +1,342 @@
+      { name: 'c', url: 'c.com' },
+    ],
+  })
+  .then(() => api.get(`/v1/samples/${name}`).set('Authorization', token));
+
+describe('tests/publish/samples.js >', () => {
+  let token;
+  let subscriber;
+  let subscribeTracker = [];
+
+  before((done) => {
+    before(() => tu.toggleOverride('enableRedisSampleStore', true));
+    subscriber = redis.createClient(DEFAULT_LOCAL_REDIS_URL);
+    subscriber.subscribe(rconf.perspectiveChannelName);
+    subscriber.on('message', (channel, msg) => subscribeTracker.push(msg));
+
+    samstoinit.eradicate()
+      .then(() => tu.createToken())
+      .then((returnedToken) => {
+        token = returnedToken;
+        done();
+      })
+      .catch(done);
+  });
+
+  after(() => tu.forceDelete(tu.db.Subject)
+    .then(() => tu.forceDelete(tu.db.Aspect))
+    .then(() => tu.forceDelete(tu.db.User))
+    .then(() => tu.forceDelete(tu.db.Profile))
+    .then(() => samstoinit.eradicate()));
+
+  afterEach(() => subscribeTracker = []);
+
+  describe('delete aspect >', () => {
+    before(() => addSubject(`${tu.namePrefix}S1`, token)
+      .then(() => addSubject(`${tu.namePrefix}S2`, token))
+      .then(() => addAspect(`${tu.namePrefix}A1`, token))
+      .then(() => upsertSample(`${tu.namePrefix}S1|${tu.namePrefix}A1`, token))
+      .then(() => upsertSample(`${tu.namePrefix}S2|${tu.namePrefix}A1`, token))
+      .then(() => (subscribeTracker = [])));
+
+    it('one sample.remove event per sample for the deleted aspect', (done) => {
+      api.delete(`/v1/aspects/${tu.namePrefix}A1`)
+        .set('Authorization', token)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(subscribeTracker).to.have.length(2);
+          const s0 = JSON.parse(subscribeTracker[0]);
+          expect(s0).to.have.property(sampleEvents.del);
+          const s0Body = s0[sampleEvents.del];
+          expect(s0Body).to.include.keys('createdAt', 'subjectId', 'aspectId',
+            'user', 'status', 'name', 'relatedLinks', 'provider', 'updatedAt',
+            'previousStatus', 'statusChangedAt', 'aspect', 'subject',
+            'absolutePath');
+
+          const s1 = JSON.parse(subscribeTracker[1]);
+          expect(s1).to.have.property(sampleEvents.del);
+          const s1Body = s1[sampleEvents.del];
+          expect(s1Body).to.include.keys('createdAt', 'subjectId', 'aspectId',
+            'user', 'status', 'name', 'relatedLinks', 'provider', 'updatedAt',
+            'previousStatus', 'statusChangedAt', 'aspect', 'subject',
+            'absolutePath');
+          done();
+        })
+    });
+  });
+
+  describe('patch aspect >', () => {
+    before(() => addSubject(`${tu.namePrefix}S3`, token)
+      .then(() => addAspect(`${tu.namePrefix}A2`, token))
+      .then(() => upsertSample(`${tu.namePrefix}S3|${tu.namePrefix}A2`, token))
+      .then(() => (subscribeTracker = [])));
+
+    it('edit tags, sample.remove followed by sample.add', (done) => {
```

Comment:
Would be good to have another test to verify that no events are published for a patch without tags.